ui-tests/src/main/java/io/jenkins/plugins/analysis/warnings/AnalysisSummary.java

```diff
@@ -0,0 +1,342 @@
+     */
+    public AnalysisResult openNewIssues() {
+        return openLink("new", "No new link found");
+    }
+
+    /**
+     * Clicks the fixed link that opens details page with the fixed issues.
+     *
+     * @return the details page with the analysis result
+     */
+    public AnalysisResult openFixedIssues() {
+        return openLink("fixed", "No fixed link found");
+    }
+
+    /**
+     * Clicks the reference build link that opens details page with the analysis results of the reference build.
+     *
+     * @return the details page with the analysis result of the reference build
+     */
+    public AnalysisResult openReferenceBuildResults() {
+        return openLink("Reference", "No reference build link found");
+    }
+
+    private AnalysisResult openLink(final String linkText, final String errorMessage) {
+        Optional<WebElement> link = findClickableResultEntryByNamePart(linkText);
+        if (link.isPresent()) {
+            return openPage(link.get(), AnalysisResult.class);
+        }
+        throw new NoSuchElementException(errorMessage);
+    }
+
+    private <T extends PageObject> T openPage(final WebElement link, final Class<T> type) {
+        String href = link.getAttribute("href");
+        T result = newInstance(type, injector, url(href), id);
+        link.click();
+
+        return result;
+    }
+
+    /**
+     * Returns the quality gate result of this parser, if set.
+     *
+     * @return Success - if the quality gate thresholds have not been reached. Failed - otherwise.
+     */
+    public QualityGateResult getQualityGateResult() {
+        for (WebElement result : results) {
+            if (result.getText().contains("Quality gate")) {
+                return QualityGateResult.valueOf(
+                        result.findElement(by.tagName("img")).getAttribute("title").toUpperCase());
+            }
+        }
+        return QualityGateResult.INACTIVE;
+    }
+
+    /**
+     * Gets the Webelement of the reset button.
+     *
+     * @throws org.openqa.selenium.NoSuchElementException
+     *         When there is no quality gate reset button.
+     */
+    public WebElement getQualityGateResetButton() throws org.openqa.selenium.NoSuchElementException {
+        for (WebElement result : results) {
+            if (result.getText().contains("Quality gate")) {
+                return result.findElement(by.id(id + "-resetReference"));
+            }
+        }
+        throw new org.openqa.selenium.NoSuchElementException("Quality gate reset button not found");
+    }
+
+    /**
+     * Checks if the quality gate reset button is present.
+     *
+     * @return True, if quality gate reset button is present.
+     */
+    public boolean hasQualityGateResetButton() {
+        try {
+            if (getQualityGateResetButton() != null) {
+                return true;
+            }
+        } catch (org.openqa.selenium.NoSuchElementException ignored) { }
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['}' at column 72 should be alone on a line.](https://app.codacy.com/manual/uhafner/warnings-ng-plugin/pullRequest?prid=5070362)